spec/type_aliases/filesource_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+
+if Puppet::Util::Package.versioncmp(Puppet.version, '4.5.0') >= 0
+  describe 'Stdlib::Filesource' do
+    describe 'valid handling' do
+      %w[
+        https://hello.com
+        https://notcreative.org
+        https://canstillaccepthttps.co.uk
+        http://anhttp.com
+        http://runningoutofideas.gov
+        file:///hello/bla
+        file:///foo/bar.log
+        puppet:///modules/foo/bar.log
```

Comment:
You should also test `puppet://someserver/modules/foo/bar.log`

Reply:
@binford2k thanks for the pointer i didn't see that varient in the docs.  Have updeated the type pattern and spec tests